api-common/src/main/java/org/openmrs/module/metadatasharing/api/db/hibernate/CustomSessionFlushTask.java

```diff
@@ -50,7 +50,7 @@
 	 * @return the result or <code>null</code>
 	 */
 	public final T executeInManualFlushMode() {
-		Session currentSession = HibernateSessionFactory.getSessionFactory().getCurrentSession();
+		Session currentSession = HibernateSessionFactory.getSessionFactory().getHibernateSessionFactory().getCurrentSession();
```

Comment:
Please change Session to DbSession and make HibernateSessionFactory.getSessionFactory() return DbSessionFactory.